drivers/platform/surface/surface_platform_profile.c

```diff
@@ -83,7 +107,8 @@ static int convert_ssam_to_profile(struct ssam_device *sdev, enum ssam_tmp_profi
 	}
 }
 
-static int convert_profile_to_ssam(struct ssam_device *sdev, enum platform_profile_option p)
+
+static int convert_profile_to_ssam_tmp(struct ssam_device *sdev, enum platform_profile_option p)
```

Comment:
Decided to rename this and a few other things to ensure it is clear it pertains to the `tmp` profile not the `fan` profile.

Reply:
Hmm, `tmp` is kind of the name of the whole subsystem. Maybe `perf` is better (i.e. performance vs. fan profile)? To me it's still somewhat of a mystery what this thing actually does on the different platforms. If you can figure that out (or something like what it's called on Windows), we could use the name of that.

Reply:
Yeah, perf is better, agreed, I'll change that over.

> To me it's still somewhat of a mystery what this thing actually does on the different platforms. If you can figure that out (or something like what it's called on Windows), we could use the name of that.

I spent the past hour or so spelunking in various `.sys` files... I don't think the fan profiles (or the whole `tmp` category) are named anywhere, I expect they're an enum in the original headers and just evaporate into a few integers somewhere in the drivers. I also can't find the limits we see in the plots in the firmware.

Interesting bits from looking for strings though;
```
SurfaceUpdate/surfacesystemmanagementframework$ strings -n 5 SurfaceSystemManagementFrameworkDriver.sys  | grep -i fan
SmfClientControl_FanControlCoreConfigurationGet{Use fan control channels with configuration}[Manual]
SmfClientControl_FanControlCoreConfigurationGet{Use fan control channels with configuration}
```
Looks like that file contains some test routines, another one is:
```
SmfProcessing_TimerPrepareLocked.MaximumProcessingTimeUpdate{Heat up the unit so that PID starts. This may take 20 minutes. Then cool the unit down so that timer stops. After that heat it again.}[Manual]
```
I wonder if there's a way to trigger that 'action' or something from Windows.


I also tried to set the fan profile and then retrieve various registers. I actually expected CID 6 ([coefficients?](https://github.com/iwanders/nixos-surface/blob/a3196593eb51deef04f6e5152f5871b048ace30a/fan.md#cid-6-coefficients)) to change, but it doesn't, what I did notice is that CID 13 does give use a float back that changes:

```
$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 0x0e 1 0 1
$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 13 1 1 
00 00 c8 41
Little endian f32: 25.0

$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 0x0e 1 0 2
$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 13 1 1 
00 00 f0 41
Little endian f32: 30.0

$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 0x0e 1 0 3
$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 13 1 1 
00 00 00 42
Little endian f32: 32.0

$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 0x0e 1 0 4
$ /home/ivor/.nix-profile/bin/python ./ctrl.py request 5 1 13 1 1 
00 00 10 42
Little endian f32: 36.0
```

I wouldn't be surprised if this is the controllers' target temperature for one of the sensors. So _technically_ there's a way to query the fan profile, but I'm not sure how much value that provides.

Anyways, I'll go make the `tmp` to `perf` change after lunch.

Reply:
Okay, not sure if this is actually what you had in mind in hindsight, but  d385b72fc03d8929b529a83e9ad40e6fe4d99c14  and 54d09a8c1  changes from `tmp` to `perf`. It felt weird to have a `convert_profile_to_ssam_perf` function that returned a `SSAM_TMP_PROFILE_BATTERY_SAVER` enum version. So instead I replaced `TMP` with `PERF` there as well...

It also renames the device struct from `ssam_tmp_profile_device` to `ssam_platform_profile_device`, since it handles more than `tmp` now, and actually handles the platform profile, with both changes to the TMP subsystem as well as the FAN subystem?

This does make the PR significantly larger, but perhaps it clears up the functionality / naming. Thoughts very welcome, I'm happy to revert if this is too extensive, I personally was hoping for a relatively small PR here, since it feels like we're bolting on just a bit of functionality. But perhaps this is the right way, your call.

Reply:
Yeah that makes sense and I think overall will result in a cleaner/more understandable module.

Reply:
Okay, based on this I've also renamed the device node in the registry from `ssam_node_tmp_pprof` to `ssam_node_perf_profile`. The original `ssam_node_tmp_pprof_with_fan` just sounded wrong.